src/main/java/SearchEngine.java

```diff
@@ -0,0 +1,93 @@
+import strategies.impl.AllStrategyImpl;
+import strategies.impl.AnyStrategyImpl;
+import strategies.impl.NoneStrategyImpl;
+import strategies.SearchStrategy;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+
+class SearchEngine {
+
+    private final FileReader fileReader;
+    private final Scanner scanner;
+    private final LineTokenizer lineTokenizer;
+    private List<String> peopleList;
```

Comment:
Seems to me it can be final as we can’t change fileReader that provides this data. Isn’t it?

Reply:
sure, i fixed this in new version